internal/functions/compare_versions.go

```diff
@@ -0,0 +1,163 @@
+package functions
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-framework/function"
+)
+
+// Ensure the implementation satisfies the desired interfaces.
+var _ function.Function = CompareVersionsFunction{}
+
+func NewCompareVersionsFunction() function.Function {
+	return &CompareVersionsFunction{}
+}
+
+type CompareVersionsFunction struct{}
+
+func (f CompareVersionsFunction) Metadata(ctx context.Context, req function.MetadataRequest, resp *function.MetadataResponse) {
+	resp.Name = "compare_versions"
+}
+
+func (f CompareVersionsFunction) Definition(ctx context.Context, req function.DefinitionRequest, resp *function.DefinitionResponse) {
+	resp.Definition = function.Definition{
+		Summary:     "Compare two version strings",
+		Description: "Given two version strings and a comparison operator, returns a boolean result of the comparison.",
+
+		Parameters: []function.Parameter{
+			function.StringParameter{
+				Name:        "version1",
+				Description: "First version string",
+			},
+			function.StringParameter{
+				Name:        "operator",
+				Description: "Comparison operator",
```

Comment:
Can we define a validator of accepted comparison operators?

Reply:
Introduced a custom validator